src/vm/errors.rs

```diff
@@ -13,24 +13,48 @@ pub enum Error {
     Arithmetic(String),
     ParseError(String),
     RecursionDetected,
-    ExpectedListPairs
+    MaxStackDepthReached,
+    MaxContextDepthReached,
+    ListDimensionTooHigh,
+    ListTooLarge,
+    BadTypeConstruction,
+    BufferTooLarge,
+    ValueTooLarge,
+    ExpectedListPairs,
+    InvalidTypeDescription,
+    BadSymbolicRepresentation(String),
+    ReservedName(String),
+    InterpreterError(String),
+    MultiplyDefined(String)
```

Comment:
"MultiplyDefined"?  As in, defined multiple times, or something to do with the multiplication operator?

Reply:
Ah, it's an error when you use the same variable name multiple times in a variable declaration like, the function definition:

`(define (foo a a) (+ a a))`

I can rename it to VariableDefinedMultipleTimes